proxy/forward_auth.go

```diff
@@ -129,6 +133,23 @@ func (p *Proxy) startAuthN(w http.ResponseWriter, r *http.Request) error {
 		uri.Path = xfu
 	}
 
+	rawRedirectURI := uri.Query().Get(urlutil.QueryRedirectURI)
```

Comment:
Oh.. this is very subtle and slightly confusing because we match on query params values in both the go, and envoy code (but I get why we can't because of the encoding).... maybe add a comment as to why this code block would be hit?

Reply:
I wish there was a better way to do this... hmm...